### PR TITLE
[ci] Script for building M1 wheels

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This PR includes a script for building wheels for Macs with M1 processors. It roughly follows the pattern of the other scripts with a few differences. 

* Manually installs `nvm`
* Uses miniforge conda to install python/pip instead of python foundation .pkgs 
* Doesn't pin numpy (we probably shouldn't be pinning it in the other scripts either...)
* Commit detection falls back to `git` instead of erroring

All of these changes were made so that the script works on a laptop, which comes with a subset of the dependencies that the x86 buildkite image comes with.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
@gjoliver this script works on my laptop, it would be great to make sure that it works on at least one other machine (i.e. yours).

File: python/build-wheel-macos-m1.sh [hunk 1]
```diff
@@ -84,7 +79,6 @@ for ((i=0; i<${#PY_VERSIONS[@]}; ++i)); do
       TRAVIS_COMMIT=$(git rev-parse HEAD)
     fi
 
-    echo $TRAVIS_COMMIT
```
Comment (gjoliver) [inline python/build-wheel-macos-m1.sh]: why not keep this?

**Review by gjoliver — APPROVED** [review]
looks good! thanks.

**Comment by gjoliver** [comment]
also, maybe we should rename the script to build-wheel-macos-arm64.sh
but that is an extremely minor comment.

**Review by simon-mo — APPROVED** [review]
LGTM

**Comment by simon-mo** [comment]
leaving up to @wuisawesome to merge

File: python/build-wheel-macos-m1.sh (hunk 1 above)
Comment (wuisawesome) [reply python/build-wheel-macos-m1.sh]: oh i mean i guess we could, it was just an accidental debugging print statement i left in. 

(note we are already printing `echo "TRAVIS_COMMIT variable detected. ray.__commit__ will be set to $TRAVIS_COMMIT"`)

**Pull request merged**